### PR TITLE
fix memory leak

### PR DESCRIPTION
**Pull request by yikenan**
to fix this bug:
https://github.com/oatpp/oatpp/issues/445

### CODE REVIEW

File: src/oatpp-zlib/Processor.cpp [hunk 1]
```diff
@@ -63,6 +63,11 @@ DeflateEncoder::DeflateEncoder(v_buff_size bufferSize, bool gzip, v_int32 compre
 
 }
 
+DeflateEncoder::~DeflateEncoder() {
+  v_int32 res = deflateEnd(&m_zStream);
+  OATPP_ASSERT(res == Z_OK);
```
Comment (lganzzzo) [inline src/oatpp-zlib/Processor.cpp]: Let's remove the assert.
Let's just log the error instead:

```cpp
OATPP_LOGE("[oatpp::zlib::DeflateEncoder::~DeflateEncoder()]", "Error. Failed call to 'deflateEnd()'.")
```

**Review by lganzzzo — COMMENTED** [review]
Hey @yikenan ,

Thanks for the PR!
Looks great!

Please change `OATPP_ASSERT` to `OATPP_LOGE` and it's ready to merge!

File: src/oatpp-zlib/Processor.cpp (hunk 1 above)
Comment (yikenan) [reply src/oatpp-zlib/Processor.cpp]: Thanks for your suggestion. I have modified the code. 
The same problem was found in the decoder and fixed. 
Please review, thank you!

**Comment by yikenan** [comment]
I ran the test program with Valgrind and found no memory leak now.

**Comment by lganzzzo** [comment]
Merged!
Great work!

**Pull request merged**